test/functional/feature_assumeutxo.py

```diff
@@ -152,6 +148,38 @@ def test_invalid_mempool_state(self, dump_output_path):
 
         self.restart_node(2, extra_args=self.extra_args[2])
 
+    def test_snapshot_in_a_divergent_chain(self, dump_output_path):
+        # First rollback node2's chain to the pregenerated one, up to height 199
+        node = self.nodes[2]
+        current_height = node.getblockcount()
+        base_height = 199
+        for block_height in range(current_height, base_height, -1):
+            block_hash = node.getblockhash(block_height)
+            node.invalidateblock(block_hash)
+        assert_equal(node.getblockcount(), START_HEIGHT)
+
+        self.log.info(f"Check importing a snapshot where current chain-tip is not an ancestor of the snapshot block but has less work")
+        # Generate a divergent chain in node2 but with less work compared to the snapshot
+        self.generate(node, nblocks=99, sync_fun=self.no_op)
+        assert node.getblockcount() < SNAPSHOT_BASE_HEIGHT
+        # Try importing the snapshot and assert its success
+        loaded = node.loadtxoutset(dump_output_path)
+        assert_equal(loaded['coins_loaded'], SNAPSHOT_BASE_HEIGHT)
+        assert_equal(loaded['base_height'], SNAPSHOT_BASE_HEIGHT)
+
+        # Restart the node and delete the snapshot chainstate from previous test.
+        self.restart_node(2, extra_args=['-reindex-chainstate=1', *self.extra_args[2]])
+        assert_equal(node.getblockcount(), 298)
```

Comment:
instead of number `298` would it make more sense to use `SNAPSHOT_BASE_HEIGHT`?

Reply:
Actually `SNAPSHOT_BASE_HEIGHT` is 299, and here I am trying to make sure that the snapshot was deleted and therefore the node is now using it's previous chain which has less work. I changed it to `assert node.getblockcount() < SNAPSHOT_BASE_HEIGHT`  for more clarity